services/time_rounding_service.rb

```diff
@@ -0,0 +1,34 @@
+class TimeRoundingService
+
+  def initialize(object_to_adjust)
+    @adjustee = object_to_adjust
+  end
+
+  def closest_fifteen
+    truncate_seconds
+    @adjustee.start_time = nearest_fifteen_after(@adjustee.start_time)
+    @adjustee.end_time = nearest_fifteen_before(@adjustee.end_time)
+    @adjustee
+  end
+
+  private
+
+  def truncate_seconds
+    @adjustee.start_time = @adjustee.start_time.change(:usec => 0, :sec => 0)
+    @adjustee.end_time = @adjustee.end_time.change(:usec => 0, :sec => 0)
+  end
+
+  def nearest_fifteen_after time
+    minutes = time.min
+    minutes = ((minutes / 60.0 * 4).ceil / 4.0 * 60).to_i
```

Comment:
If you have 16:55, won't this return `60` for the minutes ? 
Could you add a test for that as well.

Thanks
Kevin